lib/gush/client.rb

```diff
@@ -138,35 +139,31 @@ def find_job(workflow_id, job_name)
 
     def destroy_workflow(workflow)
       connection_pool.with do |redis|
-        redis.del("gush.workflows.#{workflow.id}")
+        redis.del(workflow.key)
       end
-      workflow.jobs.each {|job| destroy_job(workflow.id, job) }
+      workflow.jobs.each { |j| destroy_job(j) }
     end
 
-    def destroy_job(workflow_id, job)
+    def destroy_job(job)
       connection_pool.with do |redis|
-        redis.del("gush.jobs.#{workflow_id}.#{job.klass}")
+        redis.del(job.key)
       end
     end
 
     def expire_workflow(workflow, ttl=nil)
       ttl = ttl || configuration.ttl
-      connection_pool.with do |redis|
-        redis.expire("gush.workflows.#{workflow.id}", ttl)
-      end
-      workflow.jobs.each {|job| expire_job(workflow.id, job, ttl) }
+      keys = [workflow.key] + workflow.jobs.collect { |j| j.key }
+      persist_or_expire *keys, ttl: ttl
```

Comment:
I don't understand why we call `redis.persist` if the ttl is nil. Doing nothing is not enough?

Reply:
[`expire`](https://redis.io/commands/expire) always takes 1 parameter…
To reset the TTL to something not expirable, you have to call [`persist`](https://redis.io/commands/persist)… :sob: 
Setting TTL to ≤0 with `expire` just drop the entry (immediate expiration)…

Reply:
Do you mean that `expire_workflow` should remove the TTL when ttl is nil? I wouldn't have expected that from the method name, but I don't know the context of how it is used.

Reply:
I do that to avoid side effect. `expire!`/`expire_workflow`/`expire_job` weren't called at all on the original version. `nil` handling preserve the standard behavior if no TTL configuration found or if we want to cancel expiration (`ttl = nil`) afterwards.